nightshift/datastore.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+
+"""
+NightShift's database schema.
+
+
+dialect+driver://username:password@host:port/database
+# psycopg2
+engine = create_engine('postgresql+psycopg2://scott:tiger@localhost/mydatabase', client_encoding='utf8')
+"""
+from datetime import datetime
+
+from sqlalchemy import (
+    Boolean,
+    Column,
+    Date,
+    DateTime,
+    ForeignKey,
+    Integer,
+    PickleType,
+    PrimaryKeyConstraint,
+    String,
+    UniqueConstraint,
+)
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import relationship
+
+
+Base = declarative_base()
+
+
+class Library(Base):
+    """
+    Library system.
+    """
+
+    __tablename__ = "library"
+
+    nid = Column(Integer, primary_key=True)
+    code = Column(String(3), unique=True)
+
+    def __repr__(self):
+        return f"<Library(nid='{self.nid}', code='{self.code}')>"
+
+
+class OutputFile(Base):
+    """
+    Output MARC file info.
+    """
+
+    __tablename__ = "output_file"
+    __table_args__ = (UniqueConstraint("handle", "libraryId"),)
+
+    nid = Column(Integer, primary_key=True)
+    libraryId = Column(Integer, ForeignKey("library.nid"), nullable=False)
+    handle = Column(String, nullable=False)
+    timestamp = Column(DateTime, default=datetime.now())
+
+    def __repr__(self):
+        return (
+            f"<OutputFile(nid='{self.nid}', "
+            f"libraryId='{self.libraryId}', "
+            f"handle='{self.handle}', "
+            f"timestamp='{self.timestamp}')>"
+        )
+
+
+class Resource(Base):
+    """
+    Resource to be upgraded info.
+    """
+
+    __tablename__ = "resource"
+    __table_args__ = (
+        PrimaryKeyConstraint("sierraId", "libraryId"),
+        {},
+    )
+
+    sierraId = Column(Integer, nullable=False)
+    libraryId = Column(Integer, ForeignKey("library.nid"), nullable=False)
+    resourceCategoryId = Column(
+        Integer, ForeignKey("resource_category.nid"), nullable=False
+    )
+
+    archived = Column(Boolean, default=False)
+    bibDate = Column(Date)
+    author = Column(String(100, collation="uft8"))
+    title = Column(String(100, collation="utf8"))
+    pubDate = Column(String)
+
+    congressNumber = Column(String)
+    controlNumber = Column(String)
+    distributorNumber = Column(String)
+    otherNumber = Column(String)
+    outputId = Column(Integer, ForeignKey("output_file.nid"))
+    sourceId = Column(Integer, ForeignKey("source_file.nid"), nullable=False)
+    srcFieldsToKeep = Column(PickleType)
+    standardNumber = Column(String)
+    statusId = Column(Integer, ForeignKey("status.nid"), nullable=False)
```

Comment:
This works, but if the statuses are known ahead of time perhaps we could make this an `enum` column. Which would simplify querying a bit and opens up some interesting options: https://www.postgresql.org/docs/10/datatype-enum.html

Reply:
That's a great suggestion.  I was not familiar with `enum` type. Will use it.

Reply:
I was wondering if the `enum` type would be a good idea for library (BPL, NYPL) data? Resource table has two primary keys since sierraId may overlap between the systems (sierraId and libraryId). Maybe in this case a regular approach is better...
Another table `ResourceCategory` could also be considered, but I'm not sure how static data will be here. Every time new type of resource is added to the process, data in the table would need to be updated. For this reason I thought it may be better to stick to more traditional approach. Not sure.